content/articles/the-complete-roadmap-to-blockchain-development/index.md

```diff
@@ -0,0 +1,116 @@
+
+In web development we have two parts:-
+
+* Front-End 
+* Back- End
+
+**Front-end**
+
+The front-end is the part where the user interacts with our application. The languages involved are HTML for the structure of the page, CSS for styling the page, and Javascript which enables the user to connect to the blockchain and display data to the user.
+
+In addition, learning a framework would enable you to build a dynamic user interface quite quickly though that's optional if you have the patience to build everything from scratch. For a suitable framework, I would highly recommend "react" which is a javascript-based framework.
+
+**Back-end**
+
+For the back end, I would recommend that you learn Node JS which is javascript but on the server-side. What makes Nose JS suitable is the  NPM package manager. It enables you to install dependencies and tools for blockchain development.
+
+ #### 3. Smart contract development
+
+ Now we reach the buzzword(smart contract). What is a smart contract? In simple terms, a smart contract is a program that lives on top of the blockchain and constitutes a set of rules agreed upon by the involved parties. After execution, if these set of rules are met, the smart contract executes itself to give an output. The program enables decentralized automation by facilitating, verifying, and enforcing the conditions of an underlying agreement.
+
+ The self-executing feature of a smart contract is what makes it very important. Smart contracts enable you to exchange anything of value while also eliminating the middle man and thus you get a fair deal. In the normal world process of buying land, you would need a lawyer, whom you pay for their services first, to obtain and notarise the land transfer documents. You would also make small payments to the local authorities before the whole process is complete. But with a smart contract, you would only pay for the land and just that. In turn, the owner would transfer ownership to you by changing the name on the land title declaring you as the new owner. The transfer of ownership is kept in a public ledger and thus you would not require middlemen such as the lawyer.
+
+ The code of a smart contract cannot be changed, which in technical terms we say is immutable. However, the data is mutable(it can be changed). Contrary to normal code, changing the data of a smart contract usually costs money depending on the complexity of the code. This is because execution requires a lot of computational power which varies linearly with the complexity of the code. Therefore, in designing a smart contract, we write as simple code as possible to reduce the execution costs incurred when you change the data of the smart contract. This is what is known as "gas optimization". Gas optimization is one of the key aspects that sets the different blockchains apart in terms of adoption by users. It is for this computational power that you pay a small fee when using a smart contract.
```

Comment:
put a space between the word 'mutable' and the bracket. Instead of using "" to show a technical term, you can bold or italicize it.